pkg/api/v1alpha1/cluster_types.go

```diff
@@ -566,6 +567,21 @@ const (
 	Kube124 KubernetesVersion = "1.24"
 )
 
+// KubeVersionGreaterOrEqual checks if kube1 version is greater or equal to the kube2 version.
+func KubeVersionGreaterOrEqual(kube1, kube2 KubernetesVersion) (bool, error) {
```

Comment:
Not sure if this should be in this package
It's not an api specific operation. What we can maybe have it's a method to convert from `KubernetesVersion` to valid semver string
And move the compare logic somhere else